test2code/src/main/kotlin/com/epam/drill/plugins/test2code/Plugin.kt

```diff
@@ -246,6 +202,22 @@ class Plugin(
         }
         logger.info { "Scanned $classCount classes" }
     }
+
+    //TODO remove after admin refactoring
+    private fun createSession(
+        sessionId: String,
+        isRealtime: Boolean = true, isGlobal: Boolean = false
+    ) {
+        if (sessions[sessionId] != null)
+            return
+        synchronized(sessionId.intern()) {
```

Comment:
I am concerned that using `.intern` here is a premature optimization
- synchronized is a part of a kotlin's internals
- it utilizes `monitorEnter` and ` monitorExit` Java intrinsics, and we have no clue on how they work under-the-hood

Reply:
intern required since synchronized employed by-reference check